aea/agent.py

```diff
@@ -276,15 +265,7 @@ def stop(self) -> None:
 
         :return: None
         """
-        self.liveness.stop()
-        if self._main_loop is not None:
-            self._main_loop.stop()
-        logger.debug("[{}]: Calling teardown method...".format(self.name))
-        self.teardown()
-
-        logger.debug("[{}]: Stopping message processing...".format(self.name))
-        self.multiplexer.disconnect()
-        logger.debug("[{}]: Stopped".format(self.name))
```

Comment:
probably still worth defining these in a methods here and let the runtime call that method

Reply:
it's too runtime dependent: sync/async so no need to stay these methods here.
it will be duplicated and mostly will be unused.

a possible solution is to create standalone agent class which will support old-style behavior